.github/workflows/prbuild.yaml

```diff
@@ -1,13 +1,110 @@
 name: Build and Test Pull Request
 
 # Trigger the workflow on push or pull request
-on: [push, pull_request]
+on: 
+  push:
+    branches-ignore:
+      - "main"
```

Comment:
Why disable this workflow for pushes to `main`? That means we're only running tests against PRs, not against what's in `main`.